src/data/parts.js

```diff
@@ -0,0 +1,82 @@
+ import { parts, cars } from "./init.js";
+import { ObjectId } from "mongodb";
+import Validator from "../utils/validator.js";
+import { DataBaseException, NotFoundException } from "../utils/exceptions.js";
+
+const createPart = async (name, price, manufacturer, sellerId, carIds) => {
+    const partData = Validator.validatePart({
+        name,
+        price,
+        manufacturer,
+        sellerId,
+        carIds
+    });
+
+    let newPart = {
+        name: partData.name,
+        price: partData.price,
+        manufacturer: partData.manufacturer,
+        sellerId: new ObjectId(partData.sellerId),
+        carIds: partData.carIds.map(id => new ObjectId(id)),
+        dateAdded: new Date()
+    };
+
+    try {
+        const partsCollection = await parts();
+        const insertInfo = await partsCollection.insertOne(newPart);
+        if (!insertInfo.acknowledged || !insertInfo.insertedId) {
+            throw new Error('Could not add part');
+        }
+        const newId = insertInfo.insertedId;
+        return await getPartById(newId.toString());
+    } catch (e) {
+        throw new DataBaseException(e.message);
+    }
+};
+
+const getPartById = async (partId) => {
+    const validPartId = Validator.validateId(partId);
+    try {
+        const partsCollection = await parts();
+        const part = await partsCollection.findOne({ _id: new ObjectId(validPartId) });
+        if (!part) throw new NotFoundException('Part not found');
+        return part;
+    } catch (e) {
+        throw new DataBaseException(e.message);

```

Comment:
use databaseExceptionHandler(e) util function instead, throw databaseexception here will override the notfoundexception 404 return.